src/ui/components/Docking.cpp

```diff
@@ -28,11 +30,9 @@ static void setFlagsForAllDockNodes(ImGuiDockNodeFlags flags) {
 void DockSpace::setLayoutType(DockingLayoutType type) {
     if (type != _layoutType) {
         _layoutType = type;
+        setNeedsRelayout(true);
         if (type == DockingLayoutType::Free) {
-            // No need to relayout, just show the titlebar so user can drag it
             setFlagsForAllDockNodes(nodeFlags());
```

Comment:
The `setFlagsForAllDockNodes(nodeFlags());` could be set be optional only when in layouting mode.